lib/geohash.ex

```diff
@@ -110,6 +110,83 @@ defmodule Geohash do
     {lat, lon}
   end
 
+  @doc ~S"""
+  Calculate adjacent/2 geohash in ordinal direction ["n","s","e","w"]
+  Deals with boundary cases when adjacent is not of the same prefix.
+
+  ##Examples
+  ```
+  iex> Geohash.adjacent("abx1","n")
+  "abx4"
+
+```
+  """
+  def adjacent("",_direction) do
+    {:error, "empty geohash"}
+  end
+  def adjacent(geohash,direction) when is_list(geohash) do
+    adjacent(to_string(geohash),direction)
+  end
+  def adjacent(geohash,direction) when is_bitstring(geohash) do
+    adjacent_aux(String.downcase(geohash),String.downcase(to_string(direction)))
+  end
```

Comment:
I think you are generally correct. This may simply be a habit mixed with external input sanitizing. I will remove it. I have read the style guide, what did I miss?